### ISSUE TITLE
Average vs Sum ops when using many nodes

### ISSUE THREAD

**Issue by talkhaldi** [horovod/horovod#2298]
**Environment:**
Framework: PyTorch
Framework version: 1.6.0
Horovod version: 0.20.0
MPI version: 4.0.3
CUDA version: 10.2
NCCL version: 2.7.8-1
Python version: 3.6.5
OS and version: Ubuntu 16.04.6
GCC version: 5.5.0

**The  question:**

According to my understanding, by default there is a difference between the following settings with the same batch size:

1) Many nodes, less gradient accumulation/local_batch per gpu
2) Less nodes, more gradient accumulation/local_batch per gpu

Because in gradient accumulation (or in the local_batch in a gpu) the gradients in the same process are accumulated (Summed), but between different processes (whether on the same node or not), they are averaged by default in Horovod.

Is my understanding correct? 
And if I want to have the same behavior, should I just use the op=Sum, then the two settings would be identical?

I feel that my experiments in setting (1) are more stable (less zigzag in loss curve) than (2), even with same batch size (512). Is there a downside of using Sum op in the DistributedOptimizer?

Also, should learning rate be scaled differently when using Sum rather than Average op? (learning_rate * hvd.size())
(Due to computational limitations, Adasum can't be used)
